examples/alloc_feature.rs

```diff
@@ -21,9 +21,15 @@ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
 */
+
+#[cfg(not(feature = "alloc"))]
```

Comment:
can you please tell me a little more about the build error that you try to fix?

Reply:
Here's what I see at 382c10c217d9e239a9923ffc5b323634bba1140a (i.e. head on main at the merge of PR #5). 

```
(base) andy@ice tar-no-std % cargo test
   Compiling bitflags v2.4.0
   Compiling regex-syntax v0.7.4
   Compiling log v0.4.20
   Compiling humantime v2.1.0
   Compiling termcolor v1.2.0
   Compiling libc v0.2.147
   Compiling memchr v2.5.0
   Compiling tar-no-std v0.2.0 (/Users/andy/code/pico/3rdparty/tar-no-std)
   Compiling aho-corasick v1.0.4
   Compiling errno v0.3.2
   Compiling rustix v0.38.9
   Compiling regex-automata v0.3.6
   Compiling is-terminal v0.4.9
   Compiling regex v1.9.3
   Compiling env_logger v0.10.0
error[E0432]: unresolved import `tar_no_std::TarArchive`
  --> examples/alloc_feature.rs:24:5
   |
24 | use tar_no_std::TarArchive;
   |     ^^^^^^^^^^^^----------
   |     |           |
   |     |           help: a similar name exists in the module: `archive`
   |     no `TarArchive` in the root
   ```